### PR TITLE
[openwrt-18.06] unbound: add UCI for forward stub and auth zone clauses

### PR DESCRIPTION
**Pull request by EricLuehrsen**
Maintainer: me
Tested: TL-Archer-C7v2 OpenWrt 18.06
Description: DNS over TLS has growing interest... I realize this was late to the boat, but it needed rest time on master. I leave it to OpenWrt members to decide if this is permitted. @jow- wanted notification (I assume review) if backport was requested. Coordinated with LuCI openwrt/luci#2045.

### CODE REVIEW

File: net/unbound/files/dnsmasq.sh [hunk 1]
```diff
@@ -23,6 +23,140 @@
...
+  config_get ip   "$cfg" ip
+  config_get name "$cfg" name
+
+
+  if [ -n "$name" -a -n "$ip" ] ; then
+    create_local_zone "$name"
+
+
+    case $ip in
+      fe80:*|169.254.*)
```
Comment (jow-) [inline net/unbound/files/dnsmasq.sh]: Do you mean to catch all link-local IPs with that `fe80:*` check? Iirc that range is defined as `fe80::/10` so it'd be something like `fe[89ab][0-9a-f]:*` but maybe I misunderstood the purpose

Comment (EricLuehrsen) [reply net/unbound/files/dnsmasq.sh]: IANA reserve is all of fe80/10 but only fe80/64 is specified. RFC4291 s2.6 Link-Local addresses are for use on a single link. Link-Local addresses have the following format: 
```
| 10 | | bits | 54 bits | 64 bits |
+----------+-------------------------+----------------------------+
|1111111010| 0 | interface ID |
+----------+-------------------------+----------------------------+
```

**Comment by EricLuehrsen** [comment]
@jow- I have prepped this to be ready as if acceptable to back port. Due to openssl-1.0.2 deficiency and 18.06 unlikely upgrade with blocking packages (big patches), I again leave it to the OpenWrt members to decide if this should be rejected.

**Comment by EricLuehrsen** [comment]
With recent memory leak symptoms observed in Unbound with TLS, it is probably good to not make it easy to setup for 18.06.

**Pull request closed without merging**